docs/coding-guidelines/javascript.md

```diff
@@ -125,54 +123,105 @@ Indentation and line breaks add readability to complex statements.
+	if ( site ) {
+		if ( site.name ) {
+			if ( site.name.length <= 50 ) {
+				return site.name;
+			} else {
+				return site.name.slice( 0, 50 ) + '…';
+			}
+		} else {
+			return site.slug;
+		}
+	} else {
+		return null;
+	}
+}
+
+
+// good
+function getSiteTitle( site ) {
+	if ( ! site ) {
+		return null;
+	}
+
+	if ( ! site.name ) {
+		return site.slug;
+	}
+
+	if ( site.name.length > 50 ) {
+		return site.name.slice( 0, 50 ) + '…';
+	}
+
+	return site.name;
+}
+```
+
 ## Blocks and Curly Braces
 
-if, else, for, while, and try blocks should always use braces, and always go on multiple lines. The opening brace should be on the same line as the function definition, the conditional, or the loop. The closing brace should be on the line directly following the last statement of the block.
+`if`, `else`, `for`, `while`, and `try` blocks should always use braces, and always go on multiple lines. The opening brace should be on the same line as the function definition, the conditional, or the loop. The closing brace should be on the line directly following the last statement of the block.
 
 ```js
-var a, b, c;
+let a, b, c;
 
 if ( myFunction() ) {
-    // Expressions
+	// Expressions
 } else if ( ( a && b ) || c ) {
-    // Expressions
+	// Expressions
 } else {
-    // Expressions
+	// Expressions
 }
 ```
 
-## Multi-line Statements
-
-When a statement is too long to fit on one line, line breaks must occur after an operator.
+When all paths of a set of `if` or `else if` statements `return` a value, do not include an `else` block.
 
 ```js
-// Bad
-var html = '<p>The sum of ' + a + ' and ' + b + ' plus ' + c
-    + ' is ' + ( a + b + c );
+// bad
+function getStatusLabel() {
+	if ( isValid() ) {
+		return 'OK';
+	} else {
+		return 'Not OK';
+	}
+}
 
-// Good
-var html = '<p>The sum of ' + a + ' and ' + b + ' plus ' + c +
-    ' is ' + ( a + b + c );
+// good
+function getStatusLabel() {
+	if ( isValid() ) {
+		return 'OK';
+	}
+
+	return 'Not OK';
+}
 ```
 
-Lines should be broken into logical groups if it improves readability, such as splitting each expression of a ternary operator onto its own line, even if both will fit on a single line.
+## Multi-line Statements
+
+When a statement is too long to fit on one line, line breaks must occur after an operator.
 
 ```js
-// Acceptable
-var baz = ( true === conditionalStatement() ) ? 'thing 1' : 'thing 2';
+let html;
 
-// Better
-var baz = firstCondition( foo ) && secondCondition( bar )
-    ? qux( foo, bar )
-    : foo;
+// Bad
+html = '<p>The sum of ' + a + ' and ' + b + ' plus ' + c
+	+ ' is ' + ( a + b + c );
+
+// Good
+html = '<p>The sum of ' + a + ' and ' + b + ' plus ' + c +
+	' is ' + ( a + b + c );
 ```
 
 When a conditional is too long to fit on one line, successive lines should be indented one extra level to distinguish them from the body.
 
 ```js
 if ( firstCondition() && secondCondition() &&
-        thirdCondition() ) {
-    doStuff();
+		thirdCondition() ) {
+	doStuff();
 }
```

Comment:
when I read lines like these I find them very confusing. what if we recommended either fitting them on a single line or splitting with one per line?

```js
if (
	firstCondition() &&
	secondCondition() &&
	thirdCondition()
) {
	doStuff();
}
```

as a benefit, it can be clearer if we are mixing logical operations like this